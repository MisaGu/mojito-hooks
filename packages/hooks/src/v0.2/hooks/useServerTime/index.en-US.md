---
nav:
  path: /hooks
  group: Mojito
---

# useServerTime

Returns the server time.

## Examples

<code src="./demo/demo1.tsx" />
