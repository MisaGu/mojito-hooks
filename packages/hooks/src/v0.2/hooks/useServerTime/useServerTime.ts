import { EMojitoQueries } from '../../domain/gql/queries';
import { BaseQueryHookProps } from '../../domain/interfaces/hooks.interface';
import { useMojitoFactory } from '../useMojitoFactory/useMojitoFactory';

export type UseServerTimeData = Date;

export type UseServerTimeReturn = ReturnType<typeof useServerTime>;

export type UseServerTimeProps = BaseQueryHookProps<UseServerTimeData>;

export function useServerTime({ options }: UseServerTimeProps = {}) {
  return useMojitoFactory({
    as: 'serverTime',
    query: EMojitoQueries.serverTime,
    options,
  });
}

// legacy function
/*

export function useServerTime(): {
  serverTime: Date;
} & ReturnType<typeof useMojito> {
  const result = useMojito({
    query: EMojitoQueries.serverTime,
  });

  return {
    serverTime: result?.data?.serverTime,
    ...result,
  };
}


*/
