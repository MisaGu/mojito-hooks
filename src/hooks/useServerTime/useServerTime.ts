import { EMojitoKey } from '../../domain/enums/state.enum';
import { ServerTimeResponse } from '../../domain/interfaces';
import { BaseQueryHookProps } from '../../domain/interfaces/hooks.interface';
import { useMojitoFactory } from '../useMojitoFactory/useMojitoFactory';

export type UseServerTimeData = ServerTimeResponse;

export type UseServerTimeReturn = ReturnType<typeof useServerTime>;

export type UseServerTimeProps = BaseQueryHookProps<ServerTimeResponse>;

export function useServerTime({ options }: UseServerTimeProps = {}) {
  return useMojitoFactory<'serverTime', UseServerTimeData>({
    as: 'serverTime',
    query: EMojitoKey.serverTime,
    options,
  });
}
