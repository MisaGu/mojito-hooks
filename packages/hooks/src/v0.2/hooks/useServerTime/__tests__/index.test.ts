import { useServerTime } from '../useServerTime';

describe('useToggle', () => {
  it('should be defined', () => {
    expect(useServerTime).toBeDefined();
  });
});
