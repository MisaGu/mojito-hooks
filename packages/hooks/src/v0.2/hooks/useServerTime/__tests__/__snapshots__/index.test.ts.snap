// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useServerTime() returns the right result when needed 1`] = `
Object {
  "error": null,
  "isLoading": false,
  "queryResult": Object {
    "dataUpdatedAt": Any<Number>,
    "errorUpdateCount": 0,
    "errorUpdatedAt": 0,
    "failureCount": 0,
    "isError": false,
    "isFetched": true,
    "isFetchedAfterMount": true,
    "isFetching": false,
    "isIdle": false,
    "isLoadingError": false,
    "isPlaceholderData": false,
    "isPreviousData": false,
    "isRefetchError": false,
    "isRefetching": false,
    "isStale": false,
    "isSuccess": true,
    "remove": [Function],
    "status": "success",
  },
  "refetch": [Function],
  "serverTime": Object {
    "serverTime": 2022-05-25T14:27:18.000Z,
  },
}
`;
